src/test/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/UserReceiptMailBuilderTest.kt

```diff
@@ -83,7 +83,8 @@ class UserReceiptMailBuilderTest {
               dateString,
               totalAmountWithFeeString,
               PspTemplate(TransactionTestUtils.PSP_BUSINESS_NAME, FeeTemplate(feeString)),
-              baseTransaction.transactionAuthorizationRequestData.authorizationRequestId,
+              Optional.ofNullable(baseTransaction.transactionAuthorizationCompletedData.rrn)
+                .orElse(baseTransaction.transactionAuthorizationRequestData.authorizationRequestId),
               baseTransaction.transactionAuthorizationCompletedData.authorizationCode,
               PaymentMethodTemplate(
```

Comment:
Add a test for both a transaction with and without RRN field valued with assertion on sent rrn value to notifications service